### DISCUSSION TITLE
Support of extended list of geometry types

### DISCUSSION THREAD

**Discussion by tfr42** [postgis/postgis-java#89] [Ideas]
As documented in issue https://github.com/postgis/postgis-java/issues/54 the support of geometry types such as Curve, LineString would be very appreciated. Currently the deegree WKB/EWKB reader uses the JTS API to retrieve a geometry from a PG/PGIS database. We would like to make use of the postgis-java API with the extended set of geometry types.

**Comment by sebasbaumh** [comment]
I just saw your post in discussions after there was no progress on this topic quite for a while.

The products I build in my professional work use arcs/curves and combinations of straight and arc lines in a lot of cases.
As mentioned in #54 I reworked the implementation of postgis-java to enable it to process arcs and at the same time I tried to remove some code for older versions of PostGIS (employing WKT instead of EWKB) and also used generics where possible.

So you could have a look at https://github.com/sebasbaumh/postgis-java-ng and see how far that will take you.
I did not keep the JTS implementation as JTS is also not capable of processing arcs and I would have to linearize the geometries to get them into JTS geometries.

Ideally I would like to see that functionality going into postgis-java - but as we use my fork already in production, I am also not afraid to just maintain and extend it when needed.
I already published it to Maven Central and enabled some security and dependency checking for the builds.

**Reply by phillipross** [reply]
Ideally we should implement one or more SPIs that provide a way to support multiple ways of mapping the types to the various class hierarchies which implement the types.  This would ideally be supporting the classic/legacy classes, the JTS classes, as well as newer hierarchies such as the one written/maintained by @sebasbaumh.  Little by little I think we are decoupling the codebase and separating the functional concerns in a way that will allow us to code these SPIs.
